apps/hubble/src/storage/stores/storageCache.ts

```
@@ -83,8 +122,19 @@ export class StorageCache {
       const set = typeToSetPostfix(message.data.type);
       const fid = message.data.fid;
       const key = makeKey(fid, set);
-      const count = this._usage.get(key) ?? 0;
-      this._usage.set(key, count + 1);
+      const count = this._counts.get(key) ?? 0;
+      this._counts.set(key, count + 1);
+
+      const tsHashResult = makeTsHash(message.data.timestamp, message.hash);
+      if (tsHashResult.isOk()) {
+        const currentEarliest = this._earliestTsHashes.get(key);
+        // todo: check direction on the compare
+        if (currentEarliest === undefined || bytesCompare(currentEarliest, tsHashResult.value) > 0) {
+          this._earliestTsHashes.set(key, tsHashResult.value);
+        }
+      } else {
```

Maybe check for `tshashResult.isErr()` above so you can return the error above rather than wrapping everything following in a conditional